components/suggestion/search_date_provider.tsx

```diff
@@ -1,18 +1,19 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
-
 import Provider from './provider.jsx';
+
 import SearchDateSuggestion from './search_date_suggestion';
 
 export default class SearchDateProvider extends Provider {
-    handlePretextChanged(pretext, resultsCallback) {
+    handlePretextChanged(pretext: string, resultsCallback: (result: {matchedPretext: string;terms: string[];items: Array<{lable: string; date: string}>;component: typeof SearchDateSuggestion}
+    ) => void) {
```

Comment:
just because of the complexity of this type of `resultsCallback` i would recommend to create a new type `ResultCallback` in this same file and use that instead.